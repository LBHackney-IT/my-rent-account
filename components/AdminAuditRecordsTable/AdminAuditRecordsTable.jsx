import React from "react";
import PropTypes from "prop-types";

const adminAuditRecordsTable = ({ adminAuditRecords }) => (
  <div style={{ overflow: "auto" }}>
    <table className="govuk-table">
      <thead className="govuk-table__head">
        <tr className="govuk-table__row">
          <th scope="col" className="govuk-table__header">
            Date
          </th>
          <th scope="col" className="govuk-table__header">
            Rent Account Number
          </th>
          <th scope="col" className="govuk-table__header">
            CSSO Login
          </th>
          <th scope="col" className="govuk-table__header">
            Action
          </th>
        </tr>
      </thead>
      <tbody className="govuk-table__body">
        {adminAuditRecords.map((record, i) => (
          <tr key={i} className="govuk-table__row">
            <td className="govuk-table__cell">
              {new Date(record.timeStamp).toLocaleDateString("en-GB")}
            </td>
            <td className="govuk-table__cell">{record.rentAccountNumber}</td>
            <td className="govuk-table__cell">
              {record.cssoLogin ? "YES" : "NO"}
            </td>
            <td className="govuk-table__cell">
              {record.auditAction.toUpperCase()}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
    {adminAuditRecords.length === 0 && (
      <div className="govuk-body">No records to show</div>
    )}
  </div>
);

adminAuditRecordsTable.propTypes = {
  adminAuditRecords: PropTypes.arrayOf(
    PropTypes.shape({
      timeStamp: PropTypes.string.isRequired,
      rentAccountNumber: PropTypes.string.isRequired,
      cssoLogin: PropTypes.bool,
      auditAction: PropTypes.string,
    }).isRequired
  ).isRequired,
};

export default adminAuditRecordsTable;
