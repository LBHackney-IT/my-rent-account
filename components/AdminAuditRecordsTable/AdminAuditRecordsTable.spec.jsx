import React from "react";
import { render } from "@testing-library/react";

import AdminAuditRecordsTable from "./AdminAuditRecordsTable";

describe("AdminAuditRecordsTable component", () => {
  const props = {
    adminAuditRecords: [
      {
        timeStamp: "2016-09-05T00:00:00",
        rentAccountNumber: "9101255702",
        cssoLogin: true,
        auditAction: "view",
      },
      {
        timeStamp: "2016-09-05T00:00:00",
        rentAccountNumber: "9104565004",
        cssoLogin: false,
        auditAction: "view",
      },
    ],
  };
  it("should render properly", () => {
    const { asFragment } = render(<AdminAuditRecordsTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
