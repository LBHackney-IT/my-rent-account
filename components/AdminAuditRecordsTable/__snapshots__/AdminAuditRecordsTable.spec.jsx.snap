// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminAuditRecordsTable component should render properly 1`] = `
<DocumentFragment>
  <div
    style="overflow: auto;"
  >
    <table
      class="govuk-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header"
            scope="col"
          >
            Date
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Rent Account Number
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            CSSO Login
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            05/09/2016
          </td>
          <td
            class="govuk-table__cell"
          >
            9101255702
          </td>
          <td
            class="govuk-table__cell"
          >
            YES
          </td>
          <td
            class="govuk-table__cell"
          >
            VIEW
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            05/09/2016
          </td>
          <td
            class="govuk-table__cell"
          >
            9104565004
          </td>
          <td
            class="govuk-table__cell"
          >
            NO
          </td>
          <td
            class="govuk-table__cell"
          >
            VIEW
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
